apps/web/src/components/forms/montitor-form.tsx

```diff
@@ -505,11 +514,16 @@ export function MonitorForm({
           )}
         />
         <div className="sm:col-span-full">
-          {/*
-           * We could think of having a 'double confirmation' one,
-           * to check if the endpoint works and approve afterwards
-           * and confirm anyways even if endpoint failed
-           */}
+          {pingFailed && (
+            <FailedPingAlertConfirmation
+              props={form.getValues()}
+              submit={onSubmit}
+              loading={isPending}
+              pingFailed={pingFailed}
+              setPingFailed={setPingFailed}
```

Comment:
It's ok. But somehow it feels off to pass the full state to the children. 

How about instead making it more explicit and do something like:
```jsx
open={pingFailed}
toggle={() => setPingFailed(val => !val)}
```

Then, you can pass the `open` prop to the `Alert` and use the `toggle()` to close or open (e.g. `handleSubmit` or `onOpenChange`).